libs/api/src/lib/memberContext.ts

```diff
@@ -694,14 +700,66 @@ export class MemberContext implements MemberContextInterface {
       throw new InternalError()
     }
 
-    const invoice = await this.renewSubscriptionForUser({subscription})
+    if (startsAt < now || paidUntil) {
+      const endsAt = paidUntil ?? getNextDateForPeriodicity(startsAt, paymentPeriodicity)
 
-    // Send subscribe mail
-    await this.sendMailForSubscriptionEvent(SubscriptionEvent.SUBSCRIBE, subscription, {})
+      const user = await this.prisma.user.findUnique({
+        where: {
+          id: subscription.userID
+        },
+        select: unselectPassword
+      })
+      const invoice = await this.prisma.invoice.create({
+        data: {
+          currency: memberPlan.currency,
+          subscriptionID: subscription.id,
+          description: `Membership from ${startsAt.toISOString()} for ${user.name || user.email}`,
+          mail: user.email,
+          dueAt: startsAt,
+          scheduledDeactivationAt: endsAt,
+          items: {
+            create: {
+              name: 'Membership',
+              description: `From ${startsAt.toISOString()} to ${endsAt.toISOString()}`,
+              amount: monthlyAmount,
+              quantity: 1
+            }
+          },
+          ...(paidUntil && {paidAt: startsAt})
+        },
+        include: {
+          items: true
+        }
+      })
 
-    return {
-      subscription,
-      invoice
+      await prisma.subscription.update({
+        where: {id: subscription.id},
+        data: {
+          periods: {
+            create: {
+              startsAt,
```

Comment:
@tomaszdurka did you ensure that the period you import is always just one times the period (month,year eg.)otherwise you will get a period that is for example 3 years is that intended?